### PR TITLE
More BigInt and Bytes methods

### LINKED ISSUE #35: Adding more functionality to our `BigInt` Library
A list of improvements that would be nice to have:

- [ ] Comparisons -  implement `greaterThan(),` `lessThan()` `equalTo`, `greaterThanOrEqualTo()` , etc.
- [ ] absolute value function - `.abs()`
- [ ] the length of the string, so `123456789123456789` would return 18 (we might have this already actually, please let me know)
- [ ]  an `isI32()` func would be nice. So if it fits in an `i32` we can convert it down to i32, but if it is too too big then we return false, and can avoid crashing
- [ ] an `isZero()` method
- [x] ~~`BigInt.toString` actually returns the string value, and does not convert to i32 first (see explanation below)~~

**BigInt.toString() notes** 
BigInt.toString() is misleading since it first converts to i32, then to string
You would think you could get the pure string of the BigInt, but you can't. If you do:

`BigInt.fromI32(4149483648).toString()` 

It will return `"-145483648"`.

I think it would be better to actually turn it directly into a string. 

In the mean time, when we make docs for `graph-ts` this quirk should be mentioned.

### LINKED ISSUE #47: Missing bytes helpers
Helpers that have been requested by Gnosis:

- Unsigned `Bytes` to `BigInt` conversion
- i32 to `Bytes` conversion (and the other way around)
- String to `Bytes` conversion

### PR DESCRIPTION
**Pull request by leodasvacas**
Resolves #35, resolves #47.

This adds the helpers requested in those issues. It also allows conversion between BigInt and i32 to be implemented without host imports.

A testing harness is added to test the AS implementations. For now it only supports one test file with one test function.

### CODE REVIEW

File: test/test.ts [hunk 1]
```diff
@@ -1,7 +1,88 @@
...
+    
+    let b_i32 = 48294181
+    let b = BigInt.fromI32(b_i32)
+    assert(b == b && b.isI32() && b.toI32() == b_i32)
+    assert(a < b && a <= b)
+    
+    a_i32 = 9292928
+    a = BigInt.fromI32(9292928)
+    assert(a == a && a.isI32() && a.toI32() == a_i32)
+    assert(a < b && a <= b)
```
Comment (Jannis) [inline test/test.ts]: It would be good to also compare neighboring numbers like `123` and `124` to make sure that a difference of just `1` is covered by `<`, `>`, `<=` and `>=`.

File: test/test.ts [hunk 2]
```diff
@@ -1,7 +1,88 @@
...
+    assert(a == a && a.isI32() && a.toI32() == a_i32)
+    assert(b < a && b <= a)
+    
+    b_i32 = 10242178
+    b = BigInt.fromI32(b_i32)
+    assert(b == b && b.isI32() && b.toI32() == b_i32)
+    assert(b < a && b <= a)
+    
+    a = BigInt.fromI32(1000)
+    b = BigInt.fromI32(900)
```
Comment (Jannis) [inline test/test.ts]: It doesn't look like we're testing all boundary cases like to/from/cmp for min/max i32 and min/max u32. Could we add assertions for those?

File: index.ts [hunk 3]
```diff
@@ -153,6 +149,28 @@ export class TypedMap<K, V> {
...
+    let self = new ByteArray(4)
+    self[0] = x as u8
+    self[1] = (x >> 8) as u8
+    self[2] = (x >> 16) as u8
+    self[3] = (x >> 24) as u8
+    return self
+  }
+  
+  static fromHexString(hex: string): Bytes {
+    // Skip possible `0x` prefix.
```
[the remark this answers is not in the document]
Comment (leodasvacas) [reply index.ts]: Added assert.

[the remark this answers is not in the document]
Comment (leodasvacas) [reply index.ts]: I didn't check if the characters are hexadecimal, I assume `parseInt` will do that, is that sufficent?

**Comment by leodasvacas** [comment]
Thanks for the review! Adressed comments.

File: index.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Jannis) [reply index.ts]: Yeah, let's assume so for now :)

**Comment by Jannis** [comment]
@leodasvacas This looks good to me. Could you also prepare a documentation update in the docs repo before we merge this?

**Comment by Jannis** [comment]
Actually, I'm ok merging this now but I'll create an issue in the docs repo to add documentation for the new functionality.

**Comment by leodasvacas** [comment]
@Jannis Ah yes good reminder, I'll do that next.
